core/storage/ipfs/impl/blockservice_impl.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/ipfs/impl/blockservice_impl.hpp"
+
+using fc::storage::ipfs::Block;
+
+namespace fc::storage::ipfs {
+  outcome::result<void> BlockServiceImpl::addBlock(const Block &block) {
+    auto result = local_storage_->set(block.getCID(), block.getContent());
+    if (result.has_error()) {
+      return BlockServiceError::ADD_BLOCK_FAILED;
+    }
+    return outcome::success();
+  }
+
+  outcome::result<bool> BlockServiceImpl::has(const CID &cid) const {
+    return local_storage_->contains(cid);
+  }
+
+  outcome::result<Block::Content> BlockServiceImpl::getBlockContent(
+      const CID &cid) const {
+    outcome::result<Block::Content> data = local_storage_->get(cid);
+    if (data.has_error()) {
+      if (data.error() == IpfsDatastoreError::NOT_FOUND) {
+        return BlockServiceError::CID_NOT_FOUND;
+      }
+      return BlockServiceError::GET_BLOCK_FAILED;
+    }
+    return data;
+  }
+
+  outcome::result<void> BlockServiceImpl::removeBlock(const CID &cid) {
+    auto result = local_storage_->remove(cid);
+    if (result.has_error()) {
+      if (result.error() == IpfsDatastoreError::NOT_FOUND) {
+        return BlockServiceError::CID_NOT_FOUND;
+      }
+      return BlockServiceError::REMOVE_BLOCK_FAILED;
```

Comment:
should we really fail if cid is missing?

Reply:
Maybe, this feature never will be used, but I prefer to indicate, than remove operation was really executed